portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,64 @@
 
 package com.google.sps.servlets;
 
+import java.util.*; 
+import com.google.gson.Gson;
 import java.io.IOException;
+import com.google.sps.data.Comment;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns "Hello Jay Saleh!". TODO: modify this file to handle comments data */
+/** Servlet that writes an list of messages as a response. */
+// TODO: Modify servlet to store values in database.
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  /** Name of input field used for author name in comments section */
+  public static final String NAME_INPUT = "name-input";
+  /** Name of input field used for comment text in comments section */
+  public static final String COMMENT_INPUT = "comment-input";
+  public static final String DEFAULT_VALUE = "";
+  public static final String REDIRECT_URL = "/html/comments.html";
+
+  /** Used to store Comments for JSON parsing */
+  private List<Comment> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, NAME_INPUT, DEFAULT_VALUE);
+    String commentText = getParameter(request, COMMENT_INPUT, DEFAULT_VALUE);
+
+    messages.add(Comment.create(name, commentText));
+    response.sendRedirect(REDIRECT_URL);
+  }
+
+  /**
+   * Returns value with {@code name} from the {@code request} form. 
+   * If the {@code name} cannot be found, return {@code defaultValue}.
+   * @param request Form sent by client
+   * @param name {@code textArea} to read content of
```

Comment:
If textArea is a class then it should be written as TextArea. Or else it looks like the name of a variable.

Reply:
In `comments.html` name can either be a `<input>` or `<textarea>`.

Modified to read, `@param name {@code <input>} or {@code <textarea>} to read content of`.

Done.